### ISSUE TITLE
`if Code.ensure_loaded?(Plug)` in Sentry.Plug has unintended implications

### ISSUE THREAD

**Issue by DavidAntaramian** [getsentry/sentry-elixir#163]
This is probably a case where more documentation is simply needed, but the `if(Code.ensure_loaded?(Plug), do:` line in the [Sentry.Plug](https://github.com/getsentry/sentry-elixir/blob/4.0.2/lib/sentry/plug.ex) module can cause builds to not include `Sentry.Plug` when Sentry is compiled individually.

Background:

At Timber, we have enabled the source code reporting for errors, and as part of that, we have this line in our build script:

```sh
MIX_ENV=prod mix deps.compile sentry --force
```

This forces Sentry to recompile, per the documentation [in the README](https://github.com/getsentry/sentry-elixir/tree/4.0.2#reporting-exceptions-with-source-code). However, with the `if(Code.ensure_loaded?(Plug))` line, the `Sentry.Plug` module was _not_ being compiled. This was because the `Plug` module is not loaded into the VM while `mix deps.compile sentry --force` is being run. Therefore the check fails, and the module is never defined.

To correct this, we now use this line instead:

```sh
MIX_ENV=prod mix deps.compile plug sentry --force
```

This will ensure Plug is loaded before Sentry is compiled. This relies on the command line tooling not changing, but it works for now.

**Comment by mitchellhenke** [comment]
Thanks for opening this issue ❤️ 

The `Code.ensure_loaded?/1` is a common pattern, but I'm not sure of any alternatives that could avoid this.  It seems that the only options would be the solution you're using now or doing something like `mix do clean, compile`?

**Comment by DavidAntaramian** [comment]
I think that `mix do clean, compile` would probably be safer, though obviously it's going to take longer to compile.

I'm wondering if it's possible to have the source collection done as a separate mix task instead of part of the build process for those of us that don't include sources. The other option is to remove source code tracking altogether.

**Comment by mitchellhenke** [comment]
closed in #171
